src/hooks/useAuth.tsx

```diff
@@ -0,0 +1,54 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import { setAuth, setSignIn, setSignOut } from "store/slices/userSlice";
+import { RootState } from "store/store";
+
+const useAuth = () => {
+  const { isAuthenticated, email, role } = useSelector(
+    (state: RootState) => state.user
+  );
+
+  const dispatch = useDispatch();
+
+  useEffect(() => {
+    const token = localStorage.getItem("token");
```

Comment:
we will not user localstorage, we will use redux-persist

Reply:
Redux Persist added, once the user is logged in it will store the token on its global state authToken and localStorage. After we reload the page, React will wait until Redux re-hydrates the store to get the protected/auth routes logic working. 
Since we're dealing with global persisted states now, I've removed the isAuthenticated variable and now I'm only checking if the authToken state is defined or undefined.